### DISCUSSION TITLE
ISEL spine for Msc - Thank you

### DISCUSSION THREAD

**Discussion by NoronhaC** [joaomlourenco/novathesis#445] [Q&A] [answered]
Hi, I have a problem with the spine, the status is on final, but still it appears like in the attachment, and I would like to make it possible like the one attached.
With 3 logos instead of one only.
All help would be helpful. Thank you
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/51036880-1bca-4469-9dd1-6dafb5be352d)
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/b9dc055e-6fe0-44bc-8d04-18ef20536bda)
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/548249a8-66ff-4452-aaad-5f85d488f5ba)

**Comment by joaomlourenco** [comment]
Which version of the template are you using?
Can you switch to the latest one please, where I believe the spine is working perfectly.
Concerning the multiple logos, that's quite easy to add once the spine compiles with no errors.

**Reply by NoronhaC** [reply]
ok Thank you,
where may I check the file version.
Thanks

**Reply by NoronhaC** [reply]
Version 7.0.2 but still doesn't appear the logo. 
How I can I make it like the other spine shared here? Thank you
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/87438704-960e-45bf-917e-80b6f8fd676f)
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/efc85081-4592-4740-a6a7-8bcd9a2a62a5)

**Comment by joaomlourenco** [comment]
The “only relevant” differences are the absence of the logos and the order “year, name, title, logo”.  The text in horizontal can't apply if the width of the book spine is to thin.

If you are picky with this, you have to ask the services how to make it for thin book spines, either make the book spine much wider than it really should be, or have the horizontal text rotated to vertical.  And in this case, how should it look like!

If your are not that picky, then let's just switch the order “name / title” and add the proper logos.

What do you want to do?

**Reply by NoronhaC** [reply]
Then let's not be so picky.
But, Iwould ask then to change the Year " Ano" and name of the masters program "Mestrado..." to horizontal please.

And add the proper logos.
Thank you
Pedro

**Reply by joaomlourenco** [reply]
@NoronhaC, having the year and name of the MSc in horizontal mode is only possible if we force the book spine to be very wide (much wider than the real width of the document when printed).  Is that it what you want?

**Reply by NoronhaC** [reply]
Yes please go ahead.
Thank you
Pedro

**Reply by joaomlourenco** [reply]
Preciso dos logos que estão na lombada.  O do ISEL já tenho, os outros dois não.  Podes enviar-me por favor?  Ou fazer upload para aqui?  Obrigado.

**Reply by NoronhaC** [reply]
Sim claro que sim. Muito Obrigado
Pedro Noronha
[](url)
![Logo_IPL](https://github.com/joaomlourenco/novathesis/assets/135113777/35d7d72e-7b3d-4109-bf9f-061e297ae872)
![logo_estesl](https://github.com/joaomlourenco/novathesis/assets/135113777/66681112-b35d-47da-abf8-00c5026b2c3c)

**Reply by NoronhaC** [reply]
Hi, Is it solved?

**Reply by joaomlourenco** [reply]
Nope. Started to work on it, but did not finish yet.  I've been busy with other high-priority stuff.
Hopefully next week.  :)

**Reply by NoronhaC** [reply]
Hi Joao,

Hope you are fine and well.
Do you have any updates?

Thank you

**Reply by joaomlourenco** [reply]
Sorry… couldn't work on it in the past week.
Will work this week in my free time.

**Comment by joaomlourenco** [comment]
Hi @NoronhaC,
Finally I was able to spend some (a lot of) hours working in the template to make it  painlessly deal with multiple logos in the spine.

Please download the latest version and in `Config/1_novathesis.tex` select `school=ipl/isel/meb`, then go to the bottom of the file and select `spine/layout=trim` and `spine/width=2cm`.

Then let me know if it fits your needs.

**Comment by NoronhaC** [comment]
Hi @joaomlourenco,

Thank you, sorry for the hours.

Can I use it in overleaf as well?

Thanks once again.
Pedro

**Reply by NoronhaC** [reply]
Only one more doubt, how can I use Arial font in Overleaf with novathesis template? 
Thanks and why does the spine appears like so?
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/247cbdbb-9e1e-460f-ba81-2f6dd256c0d0)

**Reply by joaomlourenco** [reply]
You should have 3 logos in the spine (and not just the IPL logo). Can you please check this?

**Reply by joaomlourenco** [reply]
And the red boxes were supposed to NOT be there. 
Please use v7.1.3 that I will upload until 12:00 today.

**Reply by joaomlourenco** [reply]
The spine should look like this:
![image](https://github.com/joaomlourenco/novathesis/assets/2064643/8e24745d-fe70-45a3-a376-fa3bfc2ef415)

**Reply by NoronhaC** [reply]
Thanks, 

How would you activate arial font using Overleaf? 
I prefer to use Overleaf due to the synchronization with Writefull and other packages.

Thank you
Pedro

**Reply by joaomlourenco** [reply]
You need to copy your four arial font files
```
Arial.ttf
ArialBold.ttf
ArialItalic.ttf
ArialBoldItalic.ttf
```
to `NOVAthesisFiles/FontStyles/Fonts`
and then in `Config/1_novathesis.tex` uncomment an select
```latex
\ntsetup{style/font=arial}    % You can use any of the default LaTeX style files:`
```
You must also sswitch your LateX processor to  `LuaLaTeX` or `XeLaTeX` (it will not work with `pdfLaTeX`).

**Reply by NoronhaC** [reply]
Ok I have added here, and Arial font is not appearing. 
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/a24ff0b5-f068-4f15-9b36-f3723e1fde58)
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/9758d92d-537e-42b3-809d-9c6dd97bb092)

**Reply by NoronhaC** [reply]
this is solved, sorry, changed to LuaLatex and works

**Reply by NoronhaC** [reply]
Joao, the name in spine should appear between the year and the title as in the image. 
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/31ff8c43-661a-46bb-b4e7-50e622b07763)
Can it be changed?
Regards
Pedro

**Comment by joaomlourenco** [comment, marked as answer]
Support request concluded in v7.1.3.

**Reply by NoronhaC** [reply]
Hi, still missing the logos.
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/6009fcdf-5f67-48ba-a88d-d403078f2142)
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/5403559b-1a8d-410f-b142-8e56300cd501)

**Reply by NoronhaC** [reply]
This the logos don't appear and I am using the 7.1.3 version.

Thank you
Pedro

**Reply by joaomlourenco** [reply]
Please confirm that the contents of your folder `NOVAthesisFiles/Schools/ipl/isel/images` are the same as this:
https://github.com/joaomlourenco/novathesis/tree/main/NOVAthesisFiles/Schools/ipl/isel/images

If they are, it should really work.  🧐

**Reply by NoronhaC** [reply]
As you can see, same files as in the folder, and still cannot be the spine there.
Thank you.

Also regarding the front which is not ready. 
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/6e7e2e83-ba7b-4375-aaa8-eacd9a672717)

**Reply by NoronhaC** [reply]
Here you have the front page for final version and provisional.
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/7b304203-bc9f-4b51-b7dd-4a5dd2c8b9b3)
[MEB_CapaVersaoProvisoria_TFM.docx](https://github.com/joaomlourenco/novathesis/files/12929099/MEB_CapaVersaoProvisoria_TFM.docx)
[MEB_FolhaRosto_TFM.docx](https://github.com/joaomlourenco/novathesis/files/12929100/MEB_FolhaRosto_TFM.docx)
[MEB_CapaVersaoDefinitiva_TFM_v02.docx](https://github.com/joaomlourenco/novathesis/files/12929101/MEB_CapaVersaoDefinitiva_TFM_v02.docx)

**Comment by NoronhaC** [comment]
[MEB_Regras_Escrita_TFM_v20170905.pdf](https://github.com/joaomlourenco/novathesis/files/12929157/MEB_Regras_Escrita_TFM_v20170905.pdf)
If you may add these rules as well? Thank you

**Reply by joaomlourenco** [reply]
Fill an issue with a request for "New School Support" and give the link above.  I'll eventually address it.  But won't promise to do it in the short term, as my free tie now is scarce.

**Reply by NoronhaC** [reply]
Ok, I made already the request, can you give me a date?  Since I need to deliver the thesis until the end of the month. I can pay for the time used. Regards
Pedro

**Reply by NoronhaC** [reply]
Hi joao, 
I have posted the request. Did you see it? 
Thank you

**Reply by joaomlourenco** [reply]
I did see it. :)
But my agenda is really tight now and I cannot anticipate when I'll have the time to handle your request.  It's in the queue.

**Reply by NoronhaC** [reply]
Thank you, 
May it be that the problem with the spine only happens in Overleaf?
Regards
Pedro

**Reply by NoronhaC** [reply]
Hi João, did you have anytime to look at the issue? Thank you

**Reply by NoronhaC** [reply]
Hi João, 
Did you manage to look at the issue?
Regards
Pedro

**Reply by joaomlourenco** [reply]
Which issue?
The request to add full support to MEB @ ISEL?
No time now! Maybe in Christmas period or January.

**Reply by NoronhaC** [reply]
Ok thank you

**Reply by NoronhaC** [reply]
Hi Joao, 
Can you solve the issue this week then? The spine does not show the logos of the 2 institutes. 
Thank you?

**Reply by joaomlourenco** [reply]
@NoronhaC, I just downloaded a new/fresh copy of the template, in `Config/1_novathesis.tex` changed:
```latex
\ntsetup{school=ipl/isel/meb}    % The school id
…
\ntsetup{docstatus=final}     % The status of the document
```
and did a `latexmnk -pdf template`.

It produced a spine like this:
![image](https://github.com/joaomlourenco/novathesis/assets/2064643/c6e23cfe-f379-436a-8351-78700dc5ecb0)

Can you please help me understand what is wrong/missing?

**Reply by NoronhaC** [reply]
I try to change the fields like so, and it doesn't change the spine.
Is it because of the latexmnk -pdf template? Is it because I am using overleaf?
Thank you for your input Joao.
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/59be6fa6-c59f-47f8-be5b-c35a8e30e3ec)

**Reply by NoronhaC** [reply]
rules, if possible to apply. 
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/3da629a7-9ea9-441e-b99a-7149d526f6d3)
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/af82f7c9-fb75-4585-b41f-b4f0f7d6322d)
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/0be82bba-30e7-4dd8-ad4a-05f5f541b9ec)
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/32882d04-9dcc-4010-8eaa-a4566886a1c6)
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/ef31e003-9be9-4996-bd6e-57002628995b)

**Reply by NoronhaC** [reply]
was it clear?
Thank you

**Comment by joaomlourenco** [comment]
Probably there was some bug in the version you are using.  The template spine is working in the latest version.  
See https://github.com/joaomlourenco/novathesis#with-a-remote-cloud-based-service to see how to use the latest version of the template in Overleaf.
I'll try do deal with the covers and other specs until the weekend.

**Reply by NoronhaC** [reply]
it should not be an issue from overleaf, correct? thank you

**Comment by joaomlourenco** [comment]
Please check the latest version in the repository (7.1.13) and let me know.

**Comment by NoronhaC** [comment]
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/0ea51269-7d3b-4176-bc08-9816ac91fa7b)
same issue and not spine with all the logos and logos appearing on folder.
Also, I do have co-advisors can they appear? 
<img width="1819" alt="image" src="https://github.com/joaomlourenco/novathesis/assets/135113777/7d9190e8-0c1d-4236-b423-df19acca9738">
<img width="1801" alt="image" src="https://github.com/joaomlourenco/novathesis/assets/135113777/59da7048-5745-464c-9f25-5fcb16de5baa">
I have downloaded the new version and uploaded the zip file. So I cannot see what is wrong.

**Reply by joaomlourenco** [reply]
Yes.  Update your local copy with this (new) file:
https://github.com/joaomlourenco/novathesis/raw/main/NOVAthesisFiles/Schools/ipl/isel/ipl-isel-meb-defaults.ldf

**Reply by NoronhaC** [reply]
updated in that location and here is the result:
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/e89c1d60-c241-4744-87df-32d5e0999c46)
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/e6144518-e1a5-4cb8-a94b-a7621580687e)

**Comment by NoronhaC** [comment]
<img width="839" alt="image" src="https://github.com/joaomlourenco/novathesis/assets/135113777/d60d3dc1-fdc8-41fe-82b6-ef4071c70a51">
uploaded the zip novathesis

**Reply by joaomlourenco** [reply]
What is the contents of line 11 of your file "template.tex"?
Please copy&paste here the line.

**Reply by NoronhaC** [reply]
%% Version 2023-12-23 [7.1.13]

**Reply by joaomlourenco** [reply]
Please share your Overleaf with me.

**Reply by NoronhaC** [reply]
sure, where can I share it? without being public?

**Reply by NoronhaC** [reply]
any news?

**Reply by joaomlourenco** [reply]
YESSSSS!!!!  :D
The problem is with the sub-folder name `images` that must be (capitalized) `Images` in `NOVAthesisFiles/Schools/ipl`.
I.e., `NOVAthesisFiles/Schools/ipl/isel/images` must be `NOVAthesisFiles/Schools/ipl/isel/Images`.

The new version 7.1.14 fixes this.

**Reply by NoronhaC** [reply]
perfect! thank you for changing my document.
Regards

**Reply by NoronhaC** [reply]
Thank you, why is it not possible to use minted?

**Reply by NoronhaC** [reply]
the minted package

**Reply by NoronhaC** [reply]
Hi João,
The minted packae is not working properly. is there any trick to use it?
Regards
Pedro

**Reply by joaomlourenco** [reply]
Last time I checked the minted package worked all right.
Don't expect answers/help with vague messages like the one above.  Take your time to present your problem carefully, explaining clearly what you did and what is the problem, and hopefully you'll get some help.  :(

**Reply by NoronhaC** [reply]
Hi João,

I will try to explain my issue.

I am trying to incorporate a minted code in the novathesis template and by uncomment the minted package in 5_packages there is an error in compiling and document does not appear anymore, I know you present this in here and here (see images below), but it is not clear. 

I am trying to incorporate this in the annex or appendix:
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/c07d25ae-88f5-4a14-bac6-1bc8fd916876)

When I uncomment in the packages the minted this happens:
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/ad71ca37-fe01-4367-84a7-54273e9963c4)

Should I comment minted or not? It says to use it, instead of listings, correct?
Once I let only listings, it is able to compile the file, but the "tests for minted" does not appear....
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/9bb42294-7fb3-4b8b-bb61-182319b30491)

Why is ntfpackage loaded in there with minted?? 
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/d6a94f95-66a1-497f-ae54-485de21f4b0b)

A bit of a small detail on how to use minted within your package would be of great value.
Thank you once again for your time and patience and sorry if I was not able to explain all in detail.

**Reply by joaomlourenco** [reply]
First image — you have a `\caption{…}` outside of scope (i.e., before the minted environment).  See an example in the minted manual on how to use it!
Second image — as the error message says. you **DO NOT** have the arial font available in overleaf.  You have (to download it, if you fon't have it in your computer) and upload it to Overleaf to the right folder.

**Reply by NoronhaC** [reply]
thank you, I was able to solve all the issues, however this error still appears.
Should I ignore it? It is something related to the folder of IPL. Please see below, and I am using the last version of the "novathesis".
<img width="951" alt="image" src="https://github.com/joaomlourenco/novathesis/assets/135113777/7a64cfa1-fdb2-4b06-8c38-0f034bae73ec">

**Reply by joaomlourenco** [reply]
I believe you must have activated the option "print/statement" explicitly in file `Config/1_novathesis.tex` (in line 124 or nearaby).  Just set it to false (or comment the line).

**Reply by NoronhaC** [reply]
Appears " orientadora" instead of "orientadores" for more then one supervisor, can I solve it somewhere? 
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/d45ce4a6-6745-4db8-8011-5d16c15ccd89)
Thank you

**Reply by NoronhaC** [reply]
Hi prof, can you please follow this request, thank you

**Reply by joaomlourenco** [reply]
I don't know which version you are using, and I cannot reproduce the problem.  :(

You have to add 3 advisers separately, not one adviser in three lines separate by `\\`.  Is this the problem?

**Reply by NoronhaC** [reply]
Also, do ypu know why this happens in chapter 4, since I have it uncommented? and with Annex 1 
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/7fd42758-e42c-4d79-990b-4b2fb07cd8b8)
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/38a16a62-bfaa-41a3-8164-c601e7cf8e6f)
Thank you

**Reply by NoronhaC** [reply]
OK I will do that.
To add supervisor separately by \\
I am using the latest version.

Any reason for the chapter 4 to appear in black?

**Reply by joaomlourenco** [reply]
No! I don't know! It is your text.  Your LaTeX.  You must deal with it.
I am happy to help with problems of the template, but I can't help with your LaTeX problems.  I am sorry, but you have to solve them yourself.

**Reply by NoronhaC** [reply]
OK. Thank you.

**Reply by NoronhaC** [reply]
sorry but this is from the template, I have turned on the chapter 4 in 4_files.tex, the chapter 5 works fine and the 5 appears but not in chapter 4 
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/19cf6005-ed24-4c60-9f7b-6f33e8bf0c3c)
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/dc7dc459-558a-4906-b803-b753e06e94bd)
Is it the template itself? 
Thank you
Pedro

**Reply by joaomlourenco** [reply]
No. It is not the template. 
Try to include chapter 5 instead of chapter 4, ie, include twice chapter 5 as below:
```latex
\ntaddfile{chapter} {chapter1}
\ntaddfile{chapter} {chapter 2} \ntaddfile{chapter} {chapter3}
\ntaddfile{chapter}{chapter5}
\ntaddfile{chapter}{chapter5}
```

**Reply by NoronhaC** [reply]
Dear prof, 

I need just these changes to submit, can you help me, for the formatting.

For the spine as it was above, can the author name be between the title and the year?
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/8b8af677-27d2-482f-8f7d-e88fc8f6dd89)

For the acronyms, can there be no space between the different letters? I don't know how to eliminate this break.
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/a547f357-a78e-419f-85f5-b860584a5a27)

Thank you, I have until 6th of May, was pointed today.
Regards
Pedro

**Reply by NoronhaC** [reply]
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/98bd9964-7da8-4ad0-9e8f-c0393590b832)

**Reply by NoronhaC** [reply]
Here is the version used in overleaf: Version 2024-01-16 [7.1.18].
Thank you
Pedro

**Reply by NoronhaC** [reply]
School: ISEL, Degree: Mestrado em Engenharia Biomédica
version used in overleaf: Version 2024-01-16 [7.1.18].

**Reply by NoronhaC** [reply]
I think I need to add \renewcommand{\glsgroupskip}{}

But I don't know where.

Just following the https://tex.stackexchange.com/questions/129294/how-do-i-remove-alphabetical-grouping-of-acronyms-in-the-list

Hope you can clarify if possible.

Thank you

**Reply by joaomlourenco** [reply]
To remove the space between different letters, add the following to the file `Config/5_packages.tex`:
```latex
\renewcommand{\glsgroupskip}{}
```

To reorder the stuff in the book spine, download these two files:
```
https://raw.githubusercontent.com/joaomlourenco/novathesis/main/NOVAthesisFiles/spine.tex
https://raw.githubusercontent.com/joaomlourenco/novathesis/main/NOVAthesisFiles/Schools/ipl/isel/ipl-isel-meb-defaults.ldf
```
and replace the ones in:
```
NOVAthesisFiles/spine.tex
NOVAthesisFiles/Schools/ipl/isel/ipl-isel-meb-defaults.ldf
```

Let me know if it worked for you,

**Reply by NoronhaC** [reply]
Thank you! Thanks a lot,

The acronyms worked perfectly, was indeed that.

The spine is ok, I am just not sure about the text size if it is according to the requirements here:
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/7555b2e0-04c0-452e-bce8-905a7e285689)


meaning, YEAR ( Arial, 10pt Bold Center) AUTHOR NAME (Arial, 11pt Center) TITLE ( Arial 12pt Bold Center) MASTERS ( Arial, 8pt Bold Center)

Mine appears like so, on top, but the requirements are a bit different in size letter, where can I change? Thank you, I will contribute once more, and sorry for this

**Reply by joaomlourenco** [reply]
The size is calculated automatically to fit the available size!
If your title is short, it is probably over the expected size.  But if your title is long, you cannot make the font larger and make it fit the available space.

**Reply by NoronhaC** [reply]
Ok perfect, understood. thank you
Pedro

**Comment by NoronhaC** [comment]
Sorry, just a quick question, is it possible to change the color of te index to black?
Regards
Pedro

**Reply by NoronhaC** [reply]
Sorry got it, done
